pkg/controllers/user/pipeline/upgrade/upgradeimpl.go

```diff
@@ -96,7 +96,7 @@ func (l *pipelineService) Upgrade(currentVersion string) (newVersion string, err
 		}
 
 		deployment = pipelineexecution.GetRegistryDeployment(ns)
-		if registryVersion != newJekinsVersion {
+		if registryVersion != newRegistryVersion {
```

Comment:
could you briefly describe how having newJenkins version caused the error? If it's just the matter of the block below not being called when the condition is incorrect, whenever the condition is qualified, it can happen again, no?

Reply:
I think it's not the matter of being not called, currently it always gets called since `registryVersion` would always be different than `newJekinsVersion`

Reply:
So it does look like it's the matter of being not called :) Because the error happens when the condition gets called. So we have to figure out:

* whether the same error happens when `registryVersion != newRegistryVersion`
* or it happens only when the deployment is not there, yet we are trying to upgrade it

Reply:
True :) 
Happens because namespace is not passed in alerts and pipelines (tested it by forcing code to always update)